source-dropbox/oauth2.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"strings"
+	"time"
+
+	pf "github.com/estuary/flow/go/protocols/flow"
+	"golang.org/x/oauth2"
+)
+
+func GetScopes() []string {
+	return []string{
+		"files.metadata.read",
+		"files.content.read",
+		// "account_info.read",  // Not sure if we need this
+	}
+}
+
+// Most of this code was converted from the Google Auth specification updating with
+// the info on https://developers.dropbox.com/oauth-guide
+func OAuth2Spec() *pf.OAuth2 {
+	// Docs: https://www.dropbox.com/developers/documentation/http/documentation#oauth2-authorize
+	AUTH_URL_TEMPLATE_FORMAT_STR := "https://www.dropbox.com/oauth2/authorize?token_access_type=offline&client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&response_type=code&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&state={{{ state }}}&scope={{#urlencode}}%s{{/urlencode}}"
+
+	scopes := GetScopes()
+
+	spec := pf.OAuth2{
+		Provider:                  "dropbox",
+		AuthUrlTemplate:           fmt.Sprintf(AUTH_URL_TEMPLATE_FORMAT_STR, strings.Join(scopes, " ")),
+		AccessTokenUrlTemplate:    "https://api.dropboxapi.com/oauth2/token",
+		AccessTokenHeadersJsonMap: map[string]json.RawMessage{"Content-Type": json.RawMessage(`"application/x-www-form-urlencoded"`)},
+		AccessTokenBody:           "client_id={{{ client_id }}}&client_secret={{{ client_secret }}}&code={{{ code }}}&grant_type=authorization_code&redirect_uri={{{ redirect_uri }}}",
+		AccessTokenResponseJsonMap: map[string]json.RawMessage{
+			"access_token":  json.RawMessage(`"/access_token"`),
+			"refresh_token": json.RawMessage(`"/refresh_token"`),
+			"token_type":    json.RawMessage(`"/token_type"`),
+			"expires_in":    json.RawMessage(`"/expires_in"`),
+			"scope":         json.RawMessage(`"/scope"`),
```

Comment:
We do not need nor want to have all of these extra fields as part of the config. See my changes [here](https://github.com/estuary/connectors/commit/3c421fe6ff8ac96892c291fc8d0b1482d544b57e#diff-9206f1d552e9c838a5bdd3e4d39d1c505f55b63545f622cd4161d5bb0ced9189L35-R41) for what will work.